build/build/src/engine/context.rs

```diff
@@ -216,6 +216,36 @@ impl RunContext {
         Ok(())
     }
 
+    /// During the native-image build, the engine generates arg files. This function uploads them as
+    /// artifacts on the CI, so we can inspect them later.
+    /// Note that if something goes wrong, the native image arg files may not be present.
+    async fn upload_native_image_arg_files(&self) -> Result {
+        debug!("Uploading Native Image Arg Files");
+        let engine_runner_ni_argfile =
+            self.repo_root.engine.runner.target.native_image_args_txt.path.clone();
```

Comment:
You should be able to avoid cloning and just have 
`let engine_runner_ni_argfile = &self.repo_root.engine.runner.target.native_image_args_txt.path;`
(and then `&` in map construction won't be needed)

The final `path` field is always of type `PathBuf`, so there should be no type collision.

Reply:
Clone removed in 50692c515ca4f416fec1b0bc1fcb033b0f836056